ledger/participant-integration-api/src/main/scala/platform/store/backend/h2/H2StorageBackend.scala

```diff
@@ -85,16 +86,16 @@ private[backend] object H2StorageBackend
       |when matched and pcs.deduplicate_until < {submittedAt} then
       |  update set deduplicate_until={deduplicateUntil}""".stripMargin
 
-  def upsertDeduplicationEntry(
+  override def upsertDeduplicationEntry(
       key: String,
       submittedAt: Instant,
       deduplicateUntil: Instant,
   )(connection: Connection): Int =
     SQL(SQL_INSERT_COMMAND)
       .on(
         "deduplicationKey" -> key,
-        "submittedAt" -> submittedAt,
-        "deduplicateUntil" -> deduplicateUntil,
+        "submittedAt" -> Timestamp.instantToMicros(submittedAt),
```

Comment:
and this could be with the impicit decorator
`"submittedAt" -> submittedAt.toEpochMicros,`